pandas_gbq/gbq.py

```diff
@@ -711,28 +516,79 @@ def read_gbq(query, project_id=None, index_col=None, col_order=None,
 
     """
 
-    if not project_id:
-        raise TypeError("Missing required parameter: project_id")
+    # http://stackoverflow.com/questions/1094841/reusable-library-to-get-human-readable-version-of-file-size
+    def sizeof_fmt(num, suffix='B'):
+        fmt = "%3.1f %s%s"
+        for unit in ['', 'K', 'M', 'G', 'T', 'P', 'E', 'Z']:
+            if abs(num) < 1024.0:
+                return fmt % (num, unit, suffix)
+            num /= 1024.0
+        return fmt % (num, 'Y', suffix)
 
-    if dialect not in ('legacy', 'standard'):
-        raise ValueError("'{0}' is not valid for dialect".format(dialect))
+    if private_key:
+        os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = private_key
```

Comment:
Not sure if it is nice to set environ vars, can't you just include a custom reference to the private_key, if it exists, when creating the Client on row 540?

Reply:
@thejens Fixed and verified it works with my service account json key.